### DISCUSSION TITLE
[Immunization] Upcoming Encounters Screen definition | Content Changes

### DISCUSSION THREAD

**Discussion by dubdabasoduba** [opensrp/fhircore#2144] [Ideas]
- We currently have different `ON_CLICK` options on the code.
    - These functions enable us to launch or navigation to different screens based on the configurations provided.
    - Add the `LAUNCH_PROFILE` function to
      the [ProfileViewModel.kt](https://github.com/opensrp/fhircore/blob/main/android/quest/src/main/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModel.kt).
      This would give us the capabilities to launch extract screens from the profile screen.
- Using the profile configuration define an Upcoming Encounters screen with the following characteristics
    - Heading named "Upcoming Encounters"
    - Should only fetch the tasks tied to the subject. It the case of Immunizations the subject is a Patient

**Comment by pld** [comment]
It's unclear to me why we can't do this with the existing Profile configurations, why do we need code for this?

**Reply by dubdabasoduba** [reply]
It can be used. We just need a small change in the ProfileViewModel class.

**Reply by ellykits** [reply]
@dubdabasoduba It is possible to do this with the current profile configuration with the assumption, the Encounters screen will be built like profile screen. We are already doing this in Family/Household Profile -> Individual Profile when you click any household member.

**Reply by ellykits** [reply]
This requires Action configuration, which is available on the Overflow menus, Compound Texts, Service Card and Actionable Buttons. NOTE can be adopted to any of the supported widgets.

**Reply by dubdabasoduba** [reply]
Nice! thanks for the clarification @ellykits

**Reply by pld** [reply]
Changed the title to content change, since with that update, that's all this will require
